lib/system/threadlocalstorage.nim

```diff
@@ -132,6 +132,13 @@ else:
         abi: array[56 div sizeof(clong), clong]
       ThreadVarSlot {.importc: "pthread_key_t",
                     header: "<sys/types.h>".} = distinct cuint
+  elif defined(openbsd) and defined(amd64):
+    type
+      SysThread* {.importc: "pthread_t", header: "<pthread.h>".} = object
```

Comment:
https://pubs.opengroup.org/onlinepubs/7908799/xsh/pthread.h.html


The pthread_attr_t, pthread_cond_t, pthread_condattr_t, pthread_key_t, pthread_mutex_t, pthread_mutexattr_t, pthread_once_t, pthread_rwlock_t, pthread_rwlockattr_t and pthread_t types are defined as described in <sys/types.h>.

Reply:
@Araq Yes, that is true on most systems. However, OpenBSD is a little different. Please see the below links:

- [`pthread.h`](https://github.com/openbsd/src/blob/b66614995ab119f75167daaa7755b34001836821/include/pthread.h#L95) - see the type definitions from the linked line downwards
- [A search for `pthread_key_t` in the OpenBSD source](https://github.com/openbsd/src/search?l=C&p=1&q=pthread_key_t) - note the only definition is in pthread.h`
- [`sys/types.h`](https://github.com/openbsd/src/blob/master/sys/sys/types.h) - note no references to `pthread` at all

I amde this PR after actually trying the tests on OpenBSD and having them fail due to `pthread_key_t` being undefined in `sys/types.h`. This PR is the only way to make the tests pass on OpenBSD (see the error message in the linked issue for a stacktrace and compiler output - #12135)

Reply:
Additionally, it seems FreeBSD [used to be the same, circa 2005](https://lists.freebsd.org/pipermail/freebsd-threads/2005-March/002958.html).

Reply:
@Araq what is the status of this one?